server/src/main/java/org/elasticsearch/plugins/PluginsLoader.java

```diff
@@ -518,4 +540,18 @@ protected void addOpens(String pkg, Module target) {
             addExportsService(qualifiedExports, exportsService, module.getName());
         }
     }
+
+    private static void enableNativeAccess(String mainModuleName, Set<String> modulesWithNativeAccess, Controller controller) {
+        for (var moduleName : modulesWithNativeAccess) {
+            var module = controller.layer().findModule(moduleName);
+            module.ifPresentOrElse(
+                m -> NativeAccessUtil.enableNativeAccess(controller, m),
+                () -> logger.warn(
+                    "Native access not enabled for module [{}]: not a valid module name in layer [{}]",
```

Comment:
This is interesting. In what cases would this occur? Is this to make things more forgiving if the JDK changes over time?

Reply:
This should never happen; we check module names during policy parsing (via a ModuleFinder), then we derive the map of modules with native access from there. I guess this would be better changed to an assertion.